### DISCUSSION TITLE
absolute position not working for me

### DISCUSSION THREAD

**Discussion by JDubbTX** [tailwindlabs/tailwindcss#12681] [Help] [answered]
Absolute beginner here.

Following example from [here](https://github.com/11ty/eleventy-plugin-syntaxhighlight/issues/63), trying to add a button to a code block in my blog site that sits in the top right of the pre.  However, it seems to wind up after the last bit of code.  What am I doing wrong?

<img width="954" alt="tailwind-button-issue" src="https://github.com/tailwindlabs/tailwindcss/assets/56802156/1a7aebdd-7ce5-4d48-b7ba-dbaba1d51eb3">

The classes in the instruction seem to follow the  [top-right-bottom-left documentation](https://tailwindcss.com/docs/top-right-bottom-left#placing-a-positioned-element) here with the containing pre having class **relative** and the button having classes **absolute top right**

What am I missing?

**Comment by wongjn** [comment]
I think you are missing the understanding that `top` and `right` are not Tailwind classes. You may have meant `top-0 right-0` if you wanted the element to be at the top-right of the position parent.

**Reply by JDubbTX** [reply]
I started out with top-2 and right-2.  I removed the '-2' just on a whim and it changed nothing.  I'll put them back.

**Reply by JDubbTX** [reply]
Here is my current javascript with the -2's added back, which had no affect on the button placement.  The comments at the bottom where I was messing around with other things.

```javascript
(() => {
    const code = document.querySelectorAll('pre[class*="language"]');
  
    const buttonClasses = [
      'absolute',
      'top-2',
      'right-2',
      'border',
      'rounded-md',
      'p-2',
      'invisible',
      'hover:visible',
      'hover:bg-white',
      'text-slate-800',
      'group-hover/code:visible'
    ];

    const codeClasses = [
      'absolute',
      'top',
      'left'
    ];
  
    code.forEach((pre) => {
      pre.classList.add('relative');
  
      const button = document.createElement('button');
      button.classList.add(...buttonClasses);
      button.innerHTML =
        '<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="h-4 w-4"><rect width="14" height="14" x="8" y="8" rx="2" ry="2"/><path d="M4 16c-1.1 0-2-.9-2-2V4c0-1.1.9-2 2-2h10c1.1 0 2 .9 2 2"/></svg>';
  
      button.addEventListener('click', () => {
        navigator.clipboard.writeText(pre.textContent);
      });
      // pre.firstChild.classList.add(...codeClasses);
      pre.appendChild(button);
      // pre.insertBefore(button, pre.firstChild);
      
    });
  })();
```

**Reply by wongjn** [reply]
Consider checking:
- Tailwind has compiled since you added the classes.
- The `content` file globs sufficiently cover the file that the code appears in.
- `2` is a valid value for `inset`.

**Reply by JDubbTX** [reply]
I should also add that the invisible, and hover classes, and group-hover don't seem to function either.

**Reply by wongjn** [reply]
Consider reviewing the points as discussed in https://github.com/tailwindlabs/tailwindcss/discussions/12681#discussioncomment-7984964.

**Reply by wongjn** [reply, marked as answer]
> > Consider checking:
> > 
> > * Tailwind has compiled since you added the classes.
> > * The `content` file globs sufficiently cover the file that the code appears in.
> 
> Can you explain that second point in absolute beginner terms? I don't know what the 'content' file is or what globs means.[^1]

No, I'm not sure I can explain the second point any simpler than I already have. Consider reviewing [the documentation on configuring `content`](https://tailwindcss.com/docs/content-configuration).

[^1]: [#12681 (reply in thread)](https://github.com/tailwindlabs/tailwindcss/discussions/12681#discussioncomment-7984968)

**Reply by JDubbTX** [reply]
Thank you for the link - reading that link allowed me to understand what you meant.  Adding the last line shown below in tailwind.config.js fixed the issue for me.

```
module.exports = {
	content: [
		'./src/**/*.html',
		'./src/**/*.njk',
		'./src/**/*.md',
		'./src/_data/colors.js',
		'./src/_data/structure.js',
		'./src/assets/js/*.js'
	],
```

**Comment by JDubbTX** [comment]
> Consider checking:
> 
> * Tailwind has compiled since you added the classes.
> * The `content` file globs sufficiently cover the file that the code appears in.

Can you explain that second point in absolute beginner terms?  I don't know what the 'content' file is or what globs means.

**Reply by wongjn** [reply]
Replied at https://github.com/tailwindlabs/tailwindcss/discussions/12681#discussioncomment-7984974 to keep the reply chain together.
